cassandra/connection.py

```diff
@@ -586,18 +586,23 @@ def process_io_buffer(self):
 
     @defunct_on_error
     def process_msg(self, header, body):
+        self.msg_received = True
         stream_id = header.stream
         if stream_id < 0:
             callback = None
             decoder = ProtocolHandler.decode_message
             result_metadata = None
         else:
-            callback, decoder, result_metadata = self._requests.pop(stream_id)
+            try:
```

Comment:
I haven't decided whether I like this change or not. I understand why it works, but I don't like special-casing stream handling here for exactly OTOs.